Feature/member

KIMHAKSAN opened the pull request.

KIMHAKSAN committed:
feat : Member 패키지 구조 생성 및 초기 파일 설정

KIMHAKSAN committed:
feat : Member 패키지 구조 생성 및 초기 파일 추가 생성

KIMHAKSAN committed:
feat : JWT 구현 완료
filter, util, config, exception 구현

KIMHAKSAN committed:
feat : JWT, CRUD 구현
수정 사항이 필요하지만 추가적인 개발을 위해 미리 커밋 하였습니다.

KIMHAKSAN committed:
Merge branch 'develop' into feature/member

KIMHAKSAN committed:
refactor : JWT, CRUD 구현 완료
API 명세에 맞게 다건 조회를 단건으로 변경 및 형식에 맞지 않는 여러 코드 리팩토링 완료

KIMHAKSAN committed:
Merge branch 'develop' into feature/member

KIMHAKSAN committed:
refactor : 예외처리 수정
현재 다른 경로에 제 개인 예외처리 코드가 있는데 추후 이 파일을 수정해서 합칠 예정입니다.

KIMHAKSAN committed:
refactor : CRUD 부분 수정 및 피드백 반영
피드백으로 주신 내용을 반영하였습니다.

KIMHAKSAN committed:
refactor : jwt 구조 변경
SecurityConfig를 생성하기 위해 코드를 수정하였습니다.

KIMHAKSAN committed:
feat : SecurityConfig
SecurityConfig를 생성하여 테스트 중입니다.

mikejigglypuff commented on src/main/java/org/example/easytable/member/dto/response/MemberGetResDto.java:
```diff
@@ -0,0 +1,29 @@
+		this.email = member.getEmail();
+		this.address = member.getAddress();
+		this.userType = member.getUserType().name();
+		this.createdAt = member.getCreatedAt().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
```
나중에 날짜 형식 문자열은 유틸 클래스 등에서 따로 관리하도록 수정하시면 좋을 것 같아요

mikejigglypuff reviewed (commented):
intellij에서 ctrl + alt + n으로 코드 정렬만 한번 해주시면 될 것 같습니다

Hojin02 commented on src/main/java/org/example/easytable/common/filter/JwtFilter.java:
```diff
@@ -36,56 +38,61 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
-			// 인증 정보 설정
-			System.out.println("인증된 사용자 아이디: " + memberId);
-
+			request.setAttribute("memberId", memberId);
```
UserDetails 을 이용해서 유저 정보가 필요한 컨트롤러에서 @AuthenticationPrincipal로 가져올수 있도록 하면 좋을거같습니다!

KIMHAKSAN replied:
아하 지금 확인했습니다.

KIMHAKSAN committed:
refactor : CRUD 재구성
POST맨 테스트를 통해 정상적으로 코드가 테스트 되었습니다.

KIMHAKSAN committed:
refactor : CRUD 재구성
POST맨 테스트를 통해 정상적으로 코드가 테스트 되었습니다.

KIMHAKSAN committed:
refactor : a complete reconstruction

2020-byte committed:
Merge branch 'develop' into feature/member

mikejigglypuff commented on src/main/java/org/example/easytable/member/dto/request/MemberUpdateReqDto.java:
```diff
@@ -0,0 +1,14 @@
+@NoArgsConstructor // 기본 생성자 추가
+@Getter
+public class MemberUpdateReqDto {
+	private String email;   // optional 처리하거나 null을 허용
```
Member 엔티티 클래스 email의 nullable = false 를 true로 변경할 계획인 건가요?

mikejigglypuff commented on src/main/java/org/example/easytable/member/controller/MemberAuthController.java:
```diff
@@ -0,0 +1,65 @@
+		HttpServletRequest request
+	) {
+		// Authorization 헤더에서 JWT 토큰 추출
+		String token = JwtFilter.extractToken(request);
```
인증 정보를 JwtFilter에서 추출한 뒤 SecurityContextHolder에 저장해두고 Controller 메서드 매개변수에 @AuthenticationPrincipal 적용해서 가지고 오는 방법도 있다고 알아두시면 좋을 것 같아요

2020-byte committed:
refactor: improve overall authentication and authorization handling

2020-byte committed:
Merge feature/member branch and resolve conflicts

2020-byte committed:
fix: add exception handler for @PreAuthorize access denied

- Add global exception handler to catch AccessDeniedException from @PreAuthorize
- Return 403 Forbidden with ErrorResponseDto format

seoho1 reviewed (approved):
LGTM

KIMHAKSAN reviewed (commented):
확인했습니다!
